bql/grammar/grammar_test.go

```diff
@@ -145,6 +145,18 @@ func TestAcceptByParse(t *testing.T) {
 		 from ?b where {?s "old_predicate_1"@[,] ?o1.
 				?s "old_predicate_2"@[,] ?o2.
 				?s "old_predicate_3"@[,] ?o3};`,
+		// Test Deconstruct clause.
+		`deconstruct {?s "new_predicate"@[] ?o} in ?a from ?b where {?s "old_predicate"@[,] ?o} having ?s = ?o;`,
+		`deconstruct {?s "new_predicate"@[] ?o} in ?a from ?b where {?s "old_predicate"@[,] ?o};`,
+		`deconstruct {?s ?p ?o.
+			      _:v "_subject"@[] ?s.
```

Comment:
Yep, you are right, DECONSTRUCT does not support reification. However, the reason I added support for blank node is that it is possible in the CONSTRUCT statement to explicitly have blank nodes as the subject for a triple. So while we cannot DECONSTRUCT reified blank nodes with this syntax (without knowing the random uuid), we can DECONSTRUCT explicit blank nodes with known uuids. However, I am fine removing this functionality since it could be confusing, but I think it has a valid use case.